master/buildbot/data/resultspec.py

```diff
@@ -325,18 +333,28 @@ def includeFields(d):
 
             # precompute the ordering functions and sort
             if self.order:
-                order = [(lambda a, b, k=k[1:]: nonecmp(b[k], a[k]))
-                         if k[0] == '-' else
-                         (lambda a, b, k=k: nonecmp(a[k], b[k]))
-                         for k in self.order]
-
-                def cmpFunc(a, b):
-                    for f in order:
-                        c = f(a, b)
-                        if c:
-                            return c
-                    return 0
-                data.sort(cmp=cmpFunc)
+                # Do a multi-level sort by sorting from
```

Comment:
we should probably put this in a separate function, with this comments put as a docstring